packages/ethereum-contracts/contracts/agreements/ConstantFlowAgreementV1.sol

```diff
@@ -826,30 +826,6 @@ contract ConstantFlowAgreementV1 is
         newCtx = _deleteFlow(flowVars, hasPermissions, ctx, currentContext);
     }
 
-    /// @dev IConstantFlowAgreementV1.updateFlowOperatorPermissions implementation
-    function updateFlowOperatorPermissions(
-        ISuperfluidToken token,
-        address flowOperator,
-        uint8 permissions,
-        int96 flowRateAllowance, // flowRateBudget
-        bytes calldata ctx
-    ) public override returns(bytes memory newCtx) {
-        newCtx = ctx;
-        if (!FlowOperatorDefinitions.isPermissionsClean(permissions)) revert CFA_ACL_UNCLEAN_PERMISSIONS();
-        ISuperfluid.Context memory currentContext = AgreementLibrary.authorizeTokenAccess(token, ctx);
-        // [SECURITY] NOTE: we are holding the assumption here that ctx is correct and we validate it with
-        // authorizeTokenAccess:
-        if (currentContext.msgSender == flowOperator) revert CFA_ACL_NO_SENDER_FLOW_OPERATOR();
-        if (flowRateAllowance < 0) revert CFA_ACL_NO_NEGATIVE_ALLOWANCE();
-        FlowOperatorData memory flowOperatorData;
-        flowOperatorData.permissions = permissions;
-        flowOperatorData.flowRateAllowance = flowRateAllowance;
-        bytes32 flowOperatorId = _generateFlowOperatorId(currentContext.msgSender, flowOperator);
-        token.updateAgreementData(flowOperatorId, _encodeFlowOperatorData(flowOperatorData));
-
-        emit FlowOperatorUpdated(token, currentContext.msgSender, flowOperator, permissions, flowRateAllowance);
-    }
-
     /// @dev IConstantFlowAgreementV1.increaseFlowRateAllowance implementation
     function increaseFlowRateAllowance(
```

Comment:
Shouldn't this function delegate to `increaseFlowAllowanceWithPermissions`? Same goes for `decreaseFlowRateAllowance` .

Seems to me like all the code here could be de-duplicated.